### ISSUE TITLE
Notes reverted to previous versions [Android]

### ISSUE THREAD

**Issue by luxcalor** [streetwriters/notesnook#2785]
### What happened?

A few notes I have been working on constantly for 3 or 4 days all reverted to a state they were a week back. Was working on a note, left the phone for a while, opened it again and here we go. Immediately checked other notes I've been working on and the situation was the same - they were reverted to at least a week back.

Thankfully, note history contained more recent versions, but, unfortunatelly, not always the latest. Took quite a lot of time to restore them.

This is not exactly reliable.

I have not opened the web or desktop versions of NN for the last 24 hours.

### Steps to reproduce the problem

Have no idea and don't want to try.

### Version

2.5.3

### Platform/OS

Android

**Comment by luxcalor** [comment]
![Screenshot_2023-06-21-22-32-05-413_com streetwriters notesnook-edit](https://github.com/streetwriters/notesnook/assets/76074147/78090be7-4cf1-4315-8a1f-b35423a8fa8e)
And it happened again for several notes on a freshly installed app. The marked history snapshot is where it probably reverted, because the one before it is the almost last vesion.

**Comment by luxcalor** [comment]
![Screenshot_2023-06-27-18-21-36-420_com streetwriters notesnook-edit](https://github.com/streetwriters/notesnook/assets/76074147/aaea1228-8001-42d4-8870-51b39c9c65d4)
And again.
I am on the latest versions of both desktop and mobile apps.

**Comment by thecodrr** [comment]
@luxcalor hey, we are investigating this issue. Can you answer a few questions:

1. How many devices do you use Notesnook on? When this happened did you edit this same note on another device?
2. Did you see a merge conflict on this note before?
3. Do you use the Quick Note widget to append to this note? Or do you edit the note directly using the editor?
4. The above 2 screenshots you attached, are they of the same note?
5. Did you notice this same bug on the desktop app?

**Comment by luxcalor** [comment]
Hi @thecodrr, thanks. 

> 1. How many devices do you use Notesnook on? When this happened did you edit this same note on another device?

On 2 devices: Android (mostly) and Windows (occasionally). When this happened I have opened and edited the notes on the Android device only. The computer was switched off.

BTW, every time I open the note on android it gets marked as edited, which is frustrating while I am trying to get used to the absence of a manual notes order. Just opening the note gets it on top of the list. 

> 2. Did you see a merge conflict on this note before?

No. I've been working with this particular note since I subscribed to NN in April on both Android and Windows. There were no merge conflicts or revert issues until a few updates back.

> 3. Do you use the Quick Note widget to append to this note? Or do you edit the note directly using the editor?

Yes, all the time for both. I share images, files, text to the note as well as append text using the widget directly and then edit extensively using the editor.

BTW, there is this #2771 issue on Android, that happens every single time I use Quick Note widget, share functionality or a web clipper, on the latest version (2.5.4) as well.

> 4. The above 2 screenshots you attached, are they of the same note?

Yes.

> 5. Did you notice this same bug on the desktop app?

No. But then I don't use the desktop app as extensively as the Android one. I have checked the affected notes on the desktop right now, and they seem to be intact. Surprisingly there is no global edit history, only local sessions, as far as I could find. Session history is correct, with no recorded reverts.

Android 13 MIUI 14, Windows 10 64-bit, and I am always under VPN on all devices.

**Comment by luxcalor** [comment]
![Screenshot_2023-06-29-00-28-43-209_com streetwriters notesnook-edit](https://github.com/streetwriters/notesnook/assets/76074147/f8c331a3-29de-4c57-93d5-315e83f9568a)
Happened again. The note on the screenshot is the same.
Seems to not be related to the desktop app. When I opened the note in the Android app, it was reverted and the history suggested it happened 6 hours ago. I had this exact note opened in the desktop app and it was unchanged, i.e. had all latest edits that I've made yesterday, while other notes I have edited on my phone where updated and synced. Only after a force sync did the reverted note sync to the desktop in its reverted form. 

So, the only difference between reverted notes, and the ones that are synced normally I can think of is usage of quick note widget and sharing to the note. In case of unaffected notes I only used direct editor, copy-paste and direct attachment of files. 

Hope this helps.
